luarules/gadgets.lua

```diff
@@ -439,6 +449,25 @@ function gadgetHandler:LoadGadget(filename, overridevfsmode)
 		return nil
 	end
 
+  -- Centralized registration: prefer gadget manager; fall back to GetSyncedActions via manager
```

Comment:
The gadgetManager feels very anemic right now. That might be for the best, but this file is FILLED with boilerplate code around gadgets. Maybe a separate PR targeting this one, pulling in more to this system so it's more cohesive?